boa/vyper/contract.py

```diff
@@ -387,15 +369,35 @@ def __init__(
 
         self.env.register_contract(self.address, self)
 
-        fns = {fn.name: fn for fn in self.global_ctx._function_defs}
-        # add all functions from the interface to the contract
-
+        # add all exposed functions from the interface to the contract
+        # exclude internal functions since
+        # 1. they cannot be called with self._internal_method since internal
+        #    methods are not exposed in the runtime bytecode (but are included
+        #    in the deployed bytecode).
+        # 2. [TBD] Internal methods are namespaced into self.internal_methods.
```

Comment:
on the one hand, it's more ergonomic to have private functions in the same namespace as public ones. on the other hand, it could be confusing to newcomers -- why can i call private function X from titanoboa, but not from web3?